File: src/models/instance-parameters.ts
import type { Holiday } from "./holiday";
import type { Period } from "./period";
import type { WeekFrequency } from "./week-frequency";

export type InstanceParameters = Readonly<{
  nextBusinessDay: Date
  firstMonday: Date
  firstDate: Date
  lastDate: Date

  /**
   * Allows to recognize the device for next authentications.
   */
  navigatorIdentifier: string
  version: number[]
  endings: string[]
  periods: Period[]
  holidays: Holiday[]
  weekFrequencies: Map<number, WeekFrequency>
  blocksPerDay: number
}>;

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/InstanceParameters.kt
package ink.literate.pawnote.models

import kotlinx.datetime.LocalDateTime

data class InstanceParameters(
    val nextBusinessDay: LocalDateTime,
    val firstMonday: LocalDateTime,
    val firstDate: LocalDateTime,
    val lastDate: LocalDateTime,

    /**
     * Allows to recognize the device for next authentications.
     */
    val navigatorIdentifier: String? = null,
    val version: List<Int>,
    val endings: List<String>,
    val periods: List<Period>,
    val holidays: List<Holiday>,
    val weekFrequencies: Map<Int, WeekFrequency>,
    val blocksPerDay: Int
)